runtime/src/chpl-file-utils.c

```diff
@@ -204,3 +203,58 @@ qioerr chpl_fs_remove(const char* name) {
     err = qio_mkerror_errno();
   return err;
 }
+
+qioerr chpl_fs_samefile(int* ret, qio_file_t* file1, qio_file_t* file2) {
+  qioerr err = 0;
+  struct stat f1;
+  struct stat f2;
+
+  int exitStatus = fstat(file1->fd, &f1);
+  if (exitStatus) {
+    // An error occurred.  Return it.
+    err = qio_mkerror_errno();
+    return err;
+  }
+  exitStatus = fstat(file2->fd, &f2);
+  if (exitStatus) {
+    // An error occurred.  Return it.
+    err = qio_mkerror_errno();
+  } else {
+    if (f1.st_dev == f2.st_dev && f1.st_ino == f2.st_ino) {
```

Comment:
Historically, I've avoided providing a Chapel API for fstat/stat because I'm not sure that the structure
is standardized and I didn't want to figure out which things in the structure to document.

I think we could use extern with fstat but we'd have to make a Chapel version of the stat structure...